### DISCUSSION TITLE
Once I have Z-Wave JS UI running on a Windows 10 machine, how can I have HA running on a VM connect to it?

### DISCUSSION THREAD

**Discussion by imthenachoman** [zwave-js/zwave-js-ui#2895] [Q&A] [answered]
I am running HA in a Hyper-V VM. Hyper-V does not allow passing USB devices to guest VMs.

So my thought is to use Z-Wave JS UI on my Windows 10 host and then have the HA VM connect to it.

I'm having trouble figuring out how to do that.

**Comment by robertsLando** [comment]
HA needs to connect to the zwave-js server that (when enabled on zui settings) by default will run on port 3000. 

On HA side you should provide an url to allow HA instance connect to that server, so basically the question here is how to allow Hyper-V VM to connect to host network, try to check here: https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/connect-to-network

**Comment by imthenachoman** [comment, marked as answer]
@robertsLando My HA VM can access my host.

> On HA side you should provide an url to allow HA instance connect to that server

Where? I don't see a place in HA to say connect to https://my-VM-host-running-zwave-js-ui:8091/ for Z-Wave.

Or, maybe I am missing something? Do I need to also run/install Z-Wave JS UI on my HA VM? I thought I would do everything from inside HA?

**Reply by robertsLando** [reply]
Have you installed hass zwave-js integration?

**Reply by imthenachoman** [reply]
Yes. But in the integration configuration I don't see a place to enter the URL of my host machine where the Z-Wave JS server is running.

**Reply by robertsLando** [reply]
Giving that I'm not an HA user I ask @kpine or @blhoward2 if can help you with this, I think there should be a way BTW to specify to the integration the ip of the ws server

**Reply by blhoward2** [reply]
> Yes. But in the integration configuration I don't see a place to enter the URL of my host machine where the Z-Wave JS server is running.

Are you sure you're in the integration and not the addon? Maybe you missed it. Re-add the integration without removing it and it'll prompt for it again.

**Reply by imthenachoman** [reply]
Heh. I gave up. I put HA on a baremetal NUC. I figure it'll be more reliable than a VM on my daily driver. If Windows 10 has to update and reboot, it'll muck with the automations. Thanks and sorry for the trouble!
